hibike/hibike_simulator.py

```diff
@@ -1,133 +1,144 @@
+        self.params = []
+
+        self.fake_subscription_thread = FakeSubscriptionThread(fake_uids[self.ser], 0, [], self.fake_device_queue)
+        self.fake_subscription_thread.start()
+        super().__init__()
+
+    def run(self):
+        while True:
+            instruction, args = self.fake_device_queue.get()
+            res = None
+            if instruction == "ping":
+                uid, delay, params = fake_uids[self.ser], self.delay, self.params
+                uid_to_index[uid] = self.ser
+                res = ["device_subscribed", [uid, delay, params]]
+            elif instruction == "subscribe":
+                uid, delay, params = tuple(args)
+                uid_to_index[uid] = self.ser
+                res = ["device_subscribed", [uid, delay, params]]
+
+                self.fake_subscription_thread.quit = True
+                self.fake_subscription_thread.join()
+                self.fake_subscription_thread = FakeSubscriptionThread(uid, delay, params, self.fake_device_queue)
+                self.fake_subscription_thread.start()
+            elif instruction == "device_values":
+                res = [instruction, args]
+
+            self.stateQueue.put(res)
+
+class FakeSubscriptionThread(threading.Thread):
+
+    def __init__(self, uid, delay, params, fake_device_queue):
+        self.uid = uid
+        self.delay = delay
+        self.params = params
+        self.fake_device_queue = fake_device_queue
+        self.quit = False
+        super().__init__()
+
+    def run(self):
+        if self.delay != 0:
+            while True:
+                if self.quit:
+                    return
+                time.sleep(self.delay / 1000.0)
+                param_types = [hm.paramMap[hm.uid_to_device_id(self.uid)][param][1] for param in self.params]
+                params_and_values = {}
+                for param, param_type in zip(self.params, param_types):
+                    if param_type in ("bool", ):
+                        params_and_values[param] = random.choice([True, False])
+                    elif param_type in ("uint8_t", "int8_t", "uint16_t", "int16_t", "uint32_t", "int32_t", "uint64_t", "int64_t"):
+                        params_and_values[param] = random.randrange(256)
+                    else:
+                        params_and_values[param] = random.random()
+                self.fake_device_queue.put(("device_values", [self.uid, list(params_and_values.items())]))
 
 
 #############
 ## TESTING ##
 #############
 
 if __name__ == "__main__":
-    hi = Hibike()
-    hi.subToDevices([('0x000100FFFFFFFFFFFFFFFF', 1)])
+    pipeToChild, pipeFromChild = multiprocessing.Pipe()
+    badThingsQueue = multiprocessing.Queue()
+    stateQueue = multiprocessing.Queue()
+    newProcess = multiprocessing.Process(target=hibike_process, name="hibike_sim", args=[badThingsQueue, stateQueue, pipeFromChild])
+    newProcess.daemon = True
+    newProcess.start()
+    pipeToChild.send(["ready", []])
```

Comment:
Again, what's the situation with the ready message, which is supposedly deprecated?